### DISCUSSION TITLE
How to change the font

### DISCUSSION THREAD

**Discussion by vindarel** [lem-project/lem#1510] [Show and tell]
Example as of today (it might change / be abstracted):

```lisp
(when (typep (lem:implementation) 'lem-sdl2/sdl2:sdl2)
  (let ((font-regular #P"/home/jfaz/.local/share/fonts/Greybeard-13px.ttf")
        (font-bold #P"/home/jfaz/.local/share/fonts/Greybeard-13px-Bold.ttf"))
    (lem-sdl2/display:change-font (lem-sdl2/display:current-display)
                                  (lem-sdl2/font:make-font-config
                                   :latin-normal-file font-regular
                                   :latin-bold-file font-bold))))
```

by @jfaz1 on Discord

**Comment by g-gundam** [comment]
## NerdFonts

- https://github.com/ryanoasis/nerd-fonts
- https://nerdfonts.com/

People looking for some nice programming fonts can find a bunch of them over at NerdFonts.  It looks like they provide a shell script that can install one font at a time, but there's another utility that I actually like better, and it looks like he updated it recently.

- https://github.com/getnf/getnf

I like getnf, because you don't have to clone the nerd-fonts repo to get individual fonts.  Its interface also shows you which fonts you have and which fonts you don't out of the Nerd Fonts collection.  It's very sensible.

One last tip:  to get a list of available fonts, run this shell one-liner.

```bash
fc-list --format="%{family[0]}\n" | sort | uniq
```

If you have fzf, you could also add that to the end to help you search through the list of fonts interactively.

```bash
fc-list --format="%{family[0]}\n" | sort | uniq | fzf
```

**Comment by g-gundam** [comment]
I know a lot of us here are also Emacs users, so here's an Elisp function that lets you pick fonts in Emacs in a more sensible way than `M-x set-frame-font`.  It wraps `set-frame-font` and lets you pick a font from a readable list, and then it lets you pick a font size.

```lisp
(defun i/set-frame-font (&optional font size)
  "Set the font and size for the current frame."
  (interactive (list
		(completing-read
		 "font: " (process-lines "fc-list" "--format=%{family[0]}\n"))
		(completing-read
		 "size: " (mapcar (lambda (n) (format "%d" n)) (number-sequence 12 22 2)))))
  (set-frame-font (concat font " " size)))
```
